src/vm/checker/typecheck/mod.rs

```diff
@@ -350,9 +350,11 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
     fn lookup_variable(&self, name: &str, context: &TypingContext) -> TypeResult {
         if let Some(type_result) = type_reserved_variable(name) {
             Ok(type_result)
-        } else if let Some(type_result) = self.contract_context.get_variable_type(name) {
+        } else if let Some(type_result) = self.contract_context.get_constant_type(name) {
+            Ok(type_result.clone())
+        } else if let Some(type_result) = context.lookup_constant_type(name) {
             Ok(type_result.clone())
-        } else if let Some(type_result) = context.lookup_variable_type(name) {
+        } else if let Some(type_result) = self.contract_context.get_variable_type(name) {
```

Comment:
I think that this will return the type of a global variable in the following kind of case:

```
(define-data-var foo int)
(define (foo-func)
   (+ foo foo))
```

And allow that code, but we want `foo` to only be accessed through `fetch-var`.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1022/commits/1e0c21acaf4d3d7a7e0b3bfbbe4f2def4e89cc65

Reply:
Can you add a test for this failure case as well?

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1022/commits/560eb710a71d47b2d6d2c232f8b7d0decd7d35a6